tests/test_model.py

```diff
@@ -522,3 +522,417 @@ def test_model_diskonlyfilemodel_field_is_constructed_correctly(
+        dimr_config.save(filepath=temporary_save_location)
+
+        assert_files_equal(temporary_dimr_config_file, temporary_save_location)
+
+    def test_dimr_with_fmcomponent_given_correct_style_for_setting_process_for_zero(
+        self, tmp_path
+    ):
+        dimr_config_data = self.get_fm_dimr_config_data("0")
+
+        (
+            temporary_dimr_config_file,
+            temporary_save_location,
+        ) = self.setup_temporary_files(tmp_path, dimr_config_data)
+
+        dimr_config_data_expected = self.get_fm_dimr_config_data_without_process()
+        temporary_expected_dimr_config_file = tmp_path / "dimr_expected_config.xml"
+        temporary_expected_dimr_config_file.write_text(dimr_config_data_expected)
+
+        dimr_config = DIMR(filepath=temporary_dimr_config_file)
+        dimr_config.save(filepath=temporary_save_location)
+
+        assert_files_equal(temporary_expected_dimr_config_file, temporary_save_location)
+
+    @pytest.mark.parametrize(
+        "input_process, expected_process",
+        [
+            pytest.param("0", 1),
+            pytest.param("0 1", 2),
+            pytest.param("0 1 2", 3),
+            pytest.param("0 1 2 3", 4),
+            pytest.param("0 1 2 3 4", 5),
+            pytest.param("0:1", 2),
+            pytest.param("0:2", 3),
+            pytest.param("0:3", 4),
+            pytest.param("0:4", 5),
+            pytest.param("2:4", 3),
+        ],
+    )
+    def test_dimr_with_fmcomponent_process_component_set_correctly(
+        self, tmp_path, input_process, expected_process
+    ):
+        dimr_config_data = self.get_fm_dimr_config_data(input_process)
+        (
+            temporary_dimr_config_file,
+            temporary_save_location,
+        ) = self.setup_temporary_files(tmp_path, dimr_config_data)
+
+        dimr_config = DIMR(filepath=temporary_dimr_config_file)
+        dimr_config.save(filepath=temporary_save_location)
+
+        assert dimr_config.component[0].process == expected_process
+
+    def test_dimr_with_fmcomponent_given_old_incorrect_style_for_setting_process(
+        self, tmp_path
+    ):
+        process_number_single_int: int = 4
+        dimr_config_data = self.get_fm_dimr_config_data(process_number_single_int)
+        temporary_dimr_config_file, _ = self.setup_temporary_files(
+            tmp_path, dimr_config_data
+        )
+
+        with pytest.raises(ValueError) as error:
+            DIMR(filepath=temporary_dimr_config_file)
+
+        expected_message = f"In component 'test', the keyword process '{process_number_single_int}', is incorrect."
+        errormessage = str(error.value)
+        assert expected_message in errormessage
+
+    @pytest.mark.parametrize(
+        "input_process",
+        [
+            pytest.param("This is incorrect"),
+            pytest.param(1.4234),
+            pytest.param("1234556"),
+        ],
+    )
+    def test_dimr_with_fmcomponent_given_old_incorrect_style_for_setting_process2(
```

Comment:
Rather than adding a 2 at the end of the testcase name, specify the difference in the testname. E.g.:
`test_dimr_with_fmcomponent_given_invalid_type_...`

Also, why is 1234556 an invalid value? :D

Reply:
I think something was forgotten when I made the next test and used the previous test as base :)
Good catch.

1234556 is invalid because a number is given as a string and that is not supported.
Number was randomly typed, will use other number for that.